docs/en/advanced_guides/detection_transformer.md

```diff
@@ -0,0 +1,230 @@
+# Detection Transformer
+
+Detection Transformers (DETRs) are a series of Transformer-based object detectors.
+
+In most DETRs, the features extracted from backbone and neck were fed into a Transformer which is composed of an encoder and a decoder. The Transformer directly outputs a set of queries in parallel. Each query corresponds to one prediction, which may be an object or `no object`.
+
+![DETR_overall](.\DETR_overall.png)
+
+There is a brand new implementation of DETRs in MMDetection, whose features are fourfold:
+
+#### 1. A unified base detector class for DETRs.
+
+According to the existing DETRs, a unified base detector class `DetectionTransformer` is designed to standardize the architectures and forward procedures of DETRs. (mmdet/models/detectors/base_detr.py)
+
+Each detector inherited from `DetectionTransformer` is mainly consists of six components: `backbone`, `neck`, `positional_encoding`, `encoder`, `decoder`, and `bbox_head`. The `backbone` and `neck` extract features of input batch. The `positional_encoding` encodes the position of each feature points. The `encoder` processes output features of neck. The `decoder` pools encoder features of objects into the queries. At last, the `bbox_head` makes predictions and calculates losses on output queries of decoder.
+
+For the forward process, except for the earliest `extracting features` and the latest `calculating with head`, the intermediate forward process of transformer are designed as four steps: `pre_transformer`, `forward_encoder`, `pre_decoder`, and `forward_decoder`. The parameters flow among the functions are summarized as follow:
+
+![DETR_forward_process](.\DETR_forward_process.png)
```

Comment:
use a githubusercontent link instead of a png file?

Reply:
> use a githubusercontent link instead of a png file?

I'm just using local images temporarily.
The image may require adjustment.
The link will be added in the final commit.